nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,449 @@
+            If *format* is *None* and *fname* is a string, the output format is
+            deduced from the extension of the filename.
+        output_json : string, file-like or None, optional (default None)
+            Path string to a filename, or a Python file-like object.
+            The parameters of the sprite will be saved in that file
+            (unless it is None): Y and Z sizes, vmin, vmax, affine transform.
+        vmax : float, or None, optional (default None)
+            max value for mapping colors.
+        vmin : float, or None, optional (default None)
+            min value for mapping color.
+        cmap : name of a matplotlib colormap, optional (default 'Greys')
+            The colormap for the sprite. A matplotlib colormap can also
+            be passed directly in cmap.
+        threshold : a number, None, or 'auto', optional (default None)
+            If None is given, the image is not thresholded.
+            If a number is given, it is used to threshold the image:
+            values below the threshold (in absolute value) are plotted
+            as transparent. If auto is given, the threshold is determined
+            magically by analysis of the image.
+        n_colors : integer, optional (default 256)
+            The number of discrete colors to use in the colormap, if it is
+            generated.
+        format : string, optional (default 'png')
+            One of the file extensions supported by the active backend.  Most
+            backends support png, pdf, ps, eps and svg.
+        resample : boolean, optional (default True)
+            Resample to isotropic voxels, with a LR/AP/VD orientation.
+            This is necessary for proper rendering of arbitrary Niimg volumes,
+            but not necessary if the image is in an isotropic standard space.
+        interpolation : string, optional (default nearest)
+            The interpolation method for resampling
+            See nilearn.image.resample_img
+        black_bg : boolean, optional
+            If True, the background of the image is set to be black.
+
+        Returns
+        ----------
+        sprite : numpy array with the sprite
+    """
+
+    # Get cmap
+    if isinstance(cm, str):
+        cmap = plt.cm.get_cmap(cmap)
+
+    img = check_niimg_3d(img, dtype='auto')
+
+    # resample to isotropic voxel with standard orientation
+    if resample:
+        img = _resample_to_self(img, interpolation)
+
+    # Read data
+    data = _safe_get_data(img, ensure_finite=True)
+    if np.isnan(np.sum(data)):
+        data = np.nan_to_num(data)
+
+    # Deal with automatic settings of plot parameters
+    if threshold == 'auto':
+        # Threshold epsilon below a percentile value, to be sure that some
+        # voxels pass the threshold
+        threshold = fast_abs_percentile(data) - 1e-5
+
+    # threshold
+    threshold = float(threshold) if threshold is not None else None
```

Comment:
please use check_threshold from utils.param_validation

Reply:
So I checked this function. The syntax is a bit awkward, as it forces me to also import a function `percentile_func`. And as far as I understand, the behavior of `threshold` would change, as it would allow for threshold like "90%", which is not documented. I am tempted to put that off as future amelioration. What is currently implemented actually fits the doc (and plot_stat_map).

Reply:
Now I see it's included in the tests. I'll fix that.

Reply:
OK so I have added the feature, as well as a short description in the doc of `threshold`.